app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -254,6 +254,17 @@ public function getCartDataJs($checkoutType = 'multi-page')
                         ->setParentQuoteId($sessionQuote->getId())
                         ->save();
 
+                    ///////////////////////////////////////////////////////////////////////////////
+                    // if we are missing a shipping method in one-page for a non-virtual cart
+                    // then, we will return a '"needs_shipping";' string meaning that no call to
+                    // BoltCheckout.configure will be made because a shipping address is required
+                    // under these conditions in order to make that call
+                    ///////////////////////////////////////////////////////////////////////////////
+                    if ($checkoutType !== 'multi-page' && !$immutableQuote->isVirtual() && !$immutableQuote->getShippingAddress()->getShippingMethod() ) {
```

Comment:
should we bugsnag?

Reply:
@daisy1754 

It's up to you.  

Pro:  When this happens unexpectedly, there is a note in Bugsnag saying that this occurred.

Con: This expected to happen very often in refresh cases where only partial address info has been filled out and the user is filling out extra address info.  So, bugsnag would be cluttered and the reason I initially reasoned not to include it.